src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -185,23 +240,29 @@ private IServiceCollection BuildCommonServices(out AggregateException hostingSta
             var appEnvironment = PlatformServices.Default.Application;
             var contentRootPath = ResolveContentRootPath(_options.ContentRootPath, appEnvironment.ApplicationBasePath);
             var applicationName = _options.ApplicationName ?? appEnvironment.ApplicationName;
+            var hostingContext = new HostingContext();
 
             // Initialize the hosting environment
             _hostingEnvironment.Initialize(applicationName, contentRootPath, _options);
+            hostingContext.HostingEnvironment = _hostingEnvironment;
 
             var services = new ServiceCollection();
             services.AddSingleton(_hostingEnvironment);
 
+            var builder = new ConfigurationBuilder()
+                .SetBasePath(_hostingEnvironment.ContentRootPath)
+                .AddInMemoryCollection(_config.AsEnumerable());
```

Comment:
Can a reload token fire on environment variable changes? Copying this way would mean that you can't get a reload token to fire based on a hosting setting being changed in the environment. Right?

It is probably OK for now since presumably most (all?) hosting config changes would require an app restart to apply.

I guess the alternative would be to add the environment variable provider to this config, the same as the hosting config does, and then apply any UseSettings calls that have been made.